### DISCUSSION TITLE
I made a custom quirk for this device specifically TZ3000_abrsvsou - TS004F

### DISCUSSION THREAD

**Discussion by insane2subro** [zigpy/zha-device-handlers#2979] [Show and tell]
Hi, I am not a coder but just do it as a hobby. I don't know how to create a PR hence wanted to seek your help in getting it into the custom quirk for TS004F.  This quirk is better and has much more control than what was presented by default.

Here is the link to the code: https://github.com/insane2subro/zha_quirks

Supports : Mode Change from Hue / Temp Control and Normal Mode(Detect Tripple Presses for Mode Change). Single Click, Double and Toggle, Left Rotate, Right Rotate in Each of the modes independently. Press and Rotate - Left Right in Hue Mode only.

**Comment by Hedda** [comment]
If there is not already a device support request for that specific device then suggest submit a new "Device support request" under issues:

https://github.com/zigpy/zha-device-handlers/issues -> https://github.com/zigpy/zha-device-handlers/issues/new/choose

That will make it easier to search and track the progress than keeping information about it here in discussions (which almost no one reads).

Why detailed information is still needed in the template is explained in the ZHA integration document on the Home Assistant website:

https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices

PS: Also feel free to help with ZHA documentation, like  background information on why some Zigbee devices need to have a ZHA quirk:

https://www.home-assistant.io/integrations/zha#knowing-which-devices-are-supported
